subprojects/tooling-api/src/integTest/groovy/org/gradle/integtests/tooling/m5/ToolingApiIdeaModelCrossVersionSpec.groovy

```diff
@@ -215,7 +215,16 @@ project(':impl') {
         lib.scope.scope == 'TEST'
 
         IdeaModuleDependency mod = libs.find {it instanceof IdeaModuleDependency}
-        mod.dependencyModule == project.modules.find { it.name == 'api'}
+
+        def currentVersion = GradleVersion.current()
+        if (currentVersion >= GradleVersion.version("3.1")) {
+            mod.targetModuleName == project.modules.find { it.name == 'api'}.getName()
+        }
+
+        if (currentVersion.baseVersion < GradleVersion.version("4.0")) {
+            mod.dependencyModule == project.modules.find { it.name == 'api'}
+        }
+
         if (targetVersion >= GradleVersion.version("3.4")) {
```

Comment:
@adammurdoch Slightly better by using baseVersion, but before I apply this pattern to the rest of the code, please let me know what you think.  Thanks!